src/components/Tabset/index.js

```diff
@@ -88,30 +73,27 @@ export default class Tabset extends Component {
 
     updateTabsetButtonsVisibility() {
         const tabset = this.tabsetRef.current;
-        const tabsetScrollWidth = tabset.scrollWidth;
-        const tabsetScrollLeft = tabset.scrollLeft;
+        const {
+            scrollWidth: tabsetScrollWidth,
+            scrollLeft: tabsetScrollLeft,
+            offsetLeft: tabsetPosition,
+            offsetWidth: tabsetWidth,
+            children: tabsetChildren,
+        } = tabset;
         const screenWidth = window.innerWidth;
-        const tabsetPosition = tabset.offsetLeft;
-        const tabsetWidth = tabset.offsetWidth;
         const tabsetMaxScroll = tabsetScrollWidth - tabsetWidth;
-        const tabsetChildren = tabset.children;
         const tabsetChildrenArray = [].slice.call(tabsetChildren, 0);
         const tabsetChildrenWidth = tabsetChildrenArray.map(child => child.offsetWidth);
-        const tabsetChildrenTotalWidth = tabsetChildrenWidth.reduce(
-            (prev, curr) => prev + curr,
-            );
-        this.setState({ tabsetPosition });
-        this.setState({ screenWidth });
-        this.setState({ tabsetScrollLeft });
-        this.setState({ tabsetMaxScroll });
-        this.setState({ tabsetWidth });
-        if (tabsetWidth >= tabsetChildrenTotalWidth) {
-            this.setState({ isTabsetButtonsVisible: false });
-        } else {
-            this.setState({ isTabsetButtonsVisible: true });
-        }
-        console.log(tabsetWidth);
-        console.log(tabsetChildrenTotalWidth);
+        const tabsetChildrenTotalWidth = getTabsetChildrenTotalWidth(tabsetChildrenWidth);
+        const showButtons = tabsetChildrenTotalWidth > tabsetWidth;
+        this.setState({
+            tabsetPosition,
+            screenWidth,
+            tabsetScrollLeft,
+            tabsetMaxScroll,
+            tabsetWidth,
+            isTabsetButtonsVisible: showButtons,
+        });
```

Comment:
again, review this carefully, only put in state reactive things
instead use this.varibleName whenever possible
I am sure that all this properties are not needed in state